sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernatePoolDao.java

```diff
@@ -143,40 +160,104 @@ public Session currentSession() {
   }
 
   @Override
+  public Pool get(long poolId) throws IOException {
+    return withWatcherGroup((PoolImpl) currentSession().get(PoolImpl.class, poolId));
+  }
+
+  public BoxStore getBoxStore() {
+    return boxStore;
+  }
+
+  @Override
+  public Pool getByBarcode(String barcode) throws IOException {
+    if (barcode == null) throw new NullPointerException("cannot look up null barcode");
+    // This is not guaranteed to be unique, so we pick one at pseudo-random.
+    @SuppressWarnings("unchecked")
+    List<Pool> results = createCriteria().add(Restrictions.eq("identificationBarcode", barcode)).list();
+    return results.size() > 0 ? withWatcherGroup(results.get(0)) : null;
+  }
+
+  @Override
+  public Collection<Pool> getByBarcodeList(List<String> barcodeList) throws IOException {
+    Criteria criteria = createCriteria();
+    criteria.add(Restrictions.in("identificationBarcode", barcodeList));
+    @SuppressWarnings("unchecked")
+    List<Pool> results = criteria.list();
+    return withWatcherGroup(results);
+  }
+
+  public JdbcTemplate getJdbcTemplate() {
+    return jdbcTemplate;
+  }
+
+  @Override
   public Pool getPoolByBarcode(String barcode, PlatformType platformType) throws IOException {
```

Comment:
I'm confused as to why we have this method as well as `getByBarcode`. Since `identificationBarcode` is unique, shouldn't we be able to get rid of one of these?